packages/gatsby-theme-store/src/sdk/search/useSearchFiltersFromPageContext.ts

```diff
@@ -1,57 +1,53 @@
 import { useLocation } from '@reach/router'
 import { useMemo } from 'react'
 
-import type { SearchPageQueryQueryVariables } from '../../templates/__generated__/SearchPageQuery.graphql'
 import { format, parse } from './priceRange'
+import type { SearchPageContext } from '../../templates/search'
 
-// Creates a string with as many `c,c` as pathname has
-// segments.
-// For instance: cozinha/faqueiro-e-talheres would
-// generate the string c,c
-//
-// TODO: this function may have to change in the future
-const createMap = (query: string) => {
-  const splitted = query.split('/')
-
-  // We have generated all departments/brands statically, so it's safe
-  // to assume that, if the process reach this code, the path
-  // is a full text search
-  if (splitted.length === 1) {
-    return 'ft'
-  }
-
-  return new Array(splitted.length).fill('c').join(',')
+export interface SelectedFacets {
+  key: string
+  value: string
 }
 
-// I think this function should change to a more simpler version.
-const selectedFacetsAfterQueryAndMap = (query: string, map: string) => {
-  const smap = map.split(',')
-  const squery = query.split('/')
+const searchParamsFromURL = (pathname: string, params: URLSearchParams) => {
+  const smap = params.get('map')?.split(',')
+  const spath = pathname.split('/').slice(1)
 
-  const selectedFacets: Array<{ key: string; value: string }> = []
+  if (smap === undefined) {
+    throw new Error(
```

Comment:
This means that we should always have the query string `map` in the URL?

People from SEO complains a lot about the URL "pollution" caused by those `map` querystrings. 

Is there a way that we can avoid them like we do in IO, maybe using pagetype API?

Reply:
This isn't a problem in here. 

The only case that this code path is executed is if `pageContext.staticPath === false`. This means that the page wasn't pre-rendered. Since we pre-render all important pages for SEO (category tree and landing pages), if we get to this code we are either in a fullText search or in a filtered page, that have canonical/no-index markup and aren't indexed anyways